pkg/httputil/proxy/proxy.go

```diff
@@ -42,6 +42,36 @@ func getGlobalProxyConfig() *compiledConfig {
 	return cc
 }
 
+// Option returns a modified proxy round tripper.
+type Option func(base *http.Transport) *http.Transport
+
+// DialContextType is the function type of http.Transport.DialContext.
+type DialContextType func(ctx context.Context, network, addr string) (net.Conn, error)
```

Comment:
Can just name this `DialContextFunc` instead, type seems like an unfortunate name IMO.

Note: with the other proposed changes mentioned below this will likely be obsolete.